src/Evolution/DiscontinuousGalerkin/SlopeLimiters/WenoGridHelpers.hpp

```diff
@@ -0,0 +1,57 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <array>
+#include <cstddef>
+
+/// \cond
+class DataVector;
+template <size_t>
+class Direction;
+template <size_t>
+class Mesh;
+/// \endcond
+
+namespace SlopeLimiters {
+namespace Weno_detail {
+
+// Compute the coordinate positions of a neighbor element's grid points in the
+// logical coordinates of the local element.
+//
+// However, the results are organized to match the expected input of a
+// RegularGridInterpolator constructor:
+// - The coordinates are given as an array of 1D coordinate values.
+// - For any dimension where the neighbor and local element share the same 1D
+//   mesh, the array is empty. (This tells the RegularGridInterpolator that no
+//   interpolation needs to be done in this dimension.)
+//
+// NOTE: Current implementation assumes, but does not check:
+// - No h-refinement, i.e., there is only one neighbor in each direction.
```

Comment:
Doesn't this also assume an affine mapping is shared by neighbors?

Reply:
I'm not sure what you are asking; can you clarify or rephrase?

Reply:
You compute the neighbor logical coordinates by adding or subtracting 2 (if I understand the code).  This will only be true if the Element and its neighbor are related by a simple translation and the maps in each element are similar affine maps

Reply:
Yes, okay, this assumes similar affine maps between neighbors.

Depending on how this should be generalized to curved grids, and how loose that generalization can be, it may be enough that the grid deformation be continuous; this is just speculation.